torch/_refs/__init__.py

```diff
@@ -5744,24 +5744,90 @@ def log_normal(self, mean=1, std=2, generator=None):
 
 
 # TODO: add support for functionalization aten.normal_functional
+# NOTE: we cannot use the out wrapper because for the overload
+# normal.float_float_out we need to get the device and dtype from the out parameter
 @register_decomposition(aten.normal)
-@out_wrapper()
 @elementwise_type_promotion_wrapper(
-    type_promoting_args=("self",),
+    type_promoting_args=(
+        "mean",
+        "std",
+    ),
     type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
 )
-def normal(self, mean=0, std=1, generator=None):
+def normal(
+    mean=0,
+    std=1,
+    shape=None,
+    *,
+    generator=None,
+    out=None,
+    dtype=None,
+    layout=None,
+    device=None,
+    pin_memory=None,
+):
     assert generator is None
-    torch._check(std >= 0, lambda: f"normal expects std >= 0.0, but found std {std}")
+    assert layout is None
+
+    if shape is None:
+        tensors = tuple(t for t in (mean, std) if isinstance(t, TensorLike))
+        torch._check(
+            len(tensors) > 0,
+            lambda: "normal expects that either mean or std is a tensor, or shape is defined",
+        )
+        torch._check(
+            dtype is None and layout is None and device is None and pin_memory is None,
+            lambda: "Cannot pass dtype, layout, device, or pin_memory without shape",
+        )
+        if not isinstance(std, TensorLike):
+            torch._check(
+                std >= 0, lambda: f"normal expects std >= 0.0, but found std {std}"
+            )
+
+        shape = _broadcast_shapes(*(t.shape for t in tensors))
+        dtype = mean.dtype if isinstance(mean, TensorLike) else torch.float
```

Comment:
This should be 
```suggestion
        dtype = tensors[0].dtype
```
as the type cast is already performed by the decorator, so both tensors have the same dtype.